x-pack/plugin/apm-integration/src/internalClusterTest/java/org/elasticsearch/xpack/apm/ApmIT.java

```diff
@@ -68,4 +81,82 @@ public void testModule() {
+
+        assertAcked(
+            client().admin()
+                .indices()
+                .prepareCreate("test-matching")
+                .setMapping("{\"properties\":{\"message\":{\"type\":\"text\"},\"@timestamp\":{\"type\":\"date\"}}}")
+                .setSettings(
+                    Settings.builder()
+                        .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 1)
+                        .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, nodeCount * 6)
+                )
+        );
+
+        assertAcked(
+            client().admin()
+                .indices()
+                .prepareCreate("test-notmatching")
+                .setMapping("{\"properties\":{\"message\":{\"type\":\"text\"},\"@timestamp\":{\"type\":\"date\"}}}")
+                .setSettings(
+                    Settings.builder()
+                        .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 1)
+                        .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, nodeCount * 6)
+                )
+        );
+
+        ensureGreen("test-matching", "test-notmatching");
+
+        final String matchingDate = "2021-11-17";
+        final String nonMatchingDate = "2021-01-01";
+
+        final BulkRequestBuilder bulkRequestBuilder = client().prepareBulk().setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+
+        for (int i = 0; i < 1000; i++) {
+            final boolean isMatching = randomBoolean();
+            final IndexRequestBuilder indexRequestBuilder = client().prepareIndex(isMatching ? "test-matching" : "test-notmatching");
+            indexRequestBuilder.setSource(
+                "{\"@timestamp\":\"" + (isMatching ? matchingDate : nonMatchingDate) + "\",\"message\":\"\"}",
+                XContentType.JSON
+            );
+            bulkRequestBuilder.add(indexRequestBuilder);
+        }
+
+        assertFalse(bulkRequestBuilder.execute().actionGet(10, TimeUnit.SECONDS).hasFailures());
+
+        APMTracer.CAPTURING_SPAN_EXPORTER.clear();
+
+        client().prepareSearch()
+            .setQuery(new RangeQueryBuilder("@timestamp").gt("2021-11-01"))
+            .setSearchType(SearchType.QUERY_THEN_FETCH)
+            .setPreFilterShardSize(1)
+            .execute()
+            .actionGet(10, TimeUnit.SECONDS);
+
+        assertBusy(() -> {
+
+            final List<SpanData> capturedSpans = APMTracer.CAPTURING_SPAN_EXPORTER.getCapturedSpans();
+            boolean gotSearchSpan = false;
+            boolean gotCanMatchSpan = false;
+            for (SpanData capturedSpan : capturedSpans) {
+                logger.info("--> captured span [{}]", capturedSpan);
+                final String spanName = capturedSpan.getName();
+                if (spanName.equals(SearchAction.NAME)) {
+                    gotSearchSpan = true;
+                }
+                if (spanName.equals(SearchTransportService.QUERY_CAN_MATCH_NODE_NAME)) {
+                    gotCanMatchSpan = true;
+                }
+            }
```

Comment:
NIT: This could be simplified to: `APMTracer.CAPTURING_SPAN_EXPORTER.findSpanByName(..)` once https://github.com/elastic/elasticsearch/pull/80758/files is merged